src/bmson/fin_f64.rs

```diff
@@ -25,6 +25,13 @@ impl From<FinF64> for f64 {
     }
 }
 
+impl TryFrom<f64> for FinF64 {
+    type Error = ();
+    fn try_from(value: f64) -> Result<Self, Self::Error> {
+        value.is_finite().then_some(Self(value)).ok_or(())
+    }
+}
```

Comment:
It can provide an Error type like this:
```suggestion
#[derive(Debug, thiserror::Error)]
pub struct TryFromFloatError(pub(crate) ());

impl std::fmt::Display for TryFromFloatError {
    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
        write!(f, "finite number expected")
    }
}

impl TryFrom<f64> for FinF64 {
    type Error = TryFromFloatError;
    fn try_from(value: f64) -> Result<Self, Self::Error> {
        value
            .is_finite()
            .then_some(Self(value))
            .ok_or(TryFromFloatError(()))
    }
}
```